### DISCUSSION TITLE
Confused about which TP connections need to be split second shorted

### DISCUSSION THREAD

**Discussion by GHthekiwi** [lpollier/battmon#8] [Q&A] [answered]
Hello,
I use GitHub a lot for my Arduino projects but have not until today used it for more general help. And I am suitably delighted. What a spectacular tutorial lpolier has taken a lot of time to put together showing how to replace ERA-201B1 li ion cells. A great piece of work - thank you.
We have an Aibo dog that the grandkids love to bits but sadly the battery has at last died and they are heartbroken. So I am suitably motivated to see if I can put that right.
We may end up sending the cell to the U.S. for service but I am having all manner of trouble getting the dangerous goods clearance, and I don't have a confirmation from the U.S. yet that they will be able to freight it back. And it is getting very expensive. I have to drive it 8 hours north to hand it over to the freight forwarder personally so they can inspect the packaging and labeling before they will accept it.

So, I am considering options. If we can't get the cell to the U.S. and back, before we trash the Aibo I am thinking about re celling it myself. I would prefer it be done by experts but that may prove impossible.

I am all good with the tutorial but there a just a couple of things that are not 100% clear to me.

First, it looks like the fuse is indeed blown. I have not disassembled to pack but what I know from my multimeter is that the current charge is 0.99 volts (didn't surprise me) and I can get a resistance reading (about 9.91 A) between the single leg on the fuse and one side of the twin legs, and nothing between the single leg and the twin leg on the other side. And there is no connection between the two twin legs. I have ordered some SFD-047A fuses and will check with the multimeter what the readings are on a new fuse and then I will know for sure.

But the my uncertainty is about TP4/TP5 and TP 12 and TP24. I am not sure whether TP4/TP5 and simultaneously means that for resetting, the short is between both TP4 and TP5, and TP24. I can see from the multimeter that TP4 and TP5 are connected, so is it only necessary to short between either of TP4 and TP5, and TP24? I guess it is the word simultaneous that has confused me because that implies two connections simultaneously.

Also, from the photo it looks like the temporary wire to avoid blowing the (new) fuse is connected between TP5 and TP12. It doesn't look like it is also soldered to TP4. So, again does TP4/TP5 mean either of the two TPs? or is the white wire also soldered to TP4?

If someone can clarify this for me that would make re-celling an option where at the moment it isn't because of the fear of ruining the board doing the wrong thing with the reset. So, if I can't sort out the shipping (both ways) doing it myself would be an option because if I can't get it done in the U.S. I would have nothing to loose.

Hoping someone can help me here.
And once agin, thanks for the brilliant post you have put together.

**Comment by lpollier** [comment, marked as answer]
Hi GHthekiwi,

Thank you for the kind words! I’m glad the tutorial helped.

First, I can confirm that there are several companies in the U.S. and other countries offering high-quality recelling services. If you're not confident with doing the repair yourself, I highly recommend reaching out to one of these professionals. The cost is usually around $120 USD + shipping. Additionally, I know that brand new equivalent ERA-201B1 battery packs are also available, likely at a similar price.

If you decide to repair the pack yourself, you’ll need to carefully follow the guide. Start with the electrical measurements—keep in mind that resistance is measured in Ohms, not Amps, and voltage in Volts. I don't think you need to measure current during the repair process.

Regarding your questions:

**1. Fuse:**
Based on your readings, it does seem like the fuse may be blown. Checking with an ohmmeter should confirm this. Regarding your fuse reading, you mentioned:
    "I have not disassembled the pack but what I know from my multimeter is that the current charge is 0.99 volts (didn't surprise me) and I can get a resistance reading (about 9.91 A) between the single leg on the fuse and one side of the twin legs, and nothing between the single leg and the twin leg on the other side. And there is no connection between the two twin legs."

It seems you're measuring current (Amps) instead of resistance (Ohms). For a proper fuse check, you should measure the resistance between all three legs of the fuse. If none of the readings show a low resistance (close to a few Ohms), the fuse likely needs replacing. Keep in mind that fuse replacement should only be done after desoldering the cells from the PCB to avoid any shorts. Replacing the fuse also requires experience, so if you're not confident, I would recommend having a professional handle it.

**2. Wire between TP4/TP5 and TP12:**
TP4 and TP5 are connected together in the PCB layout, so you can connect either one to TP12 to prevent the fuse from blowing during the resoldering of the new cell pack to the PCB. Therefore, the wire should be connected between either TP4 or TP5 and TP12.

**3. TP4/TP5 and TP24 Short Time Connection at Resetting Step:**
For the final step of resetting the first-level battery protection IC1, briefly short either TP4 or TP5 to TP24. This will trigger the reset process.

If you want to check where the test points TP4, TP5, TP12, and TP24 are connected on the PCB, you can refer to the schematic here:
[https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_reverse_engineering_schematic.pdf](https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_reverse_engineering_schematic.pdf)

I hope that clears things up! If you’re comfortable with the process, re-celling it yourself could be a good option. Good luck with the Aibo!

Best regards,
Loïc

**Reply by GHthekiwi** [reply]
Loïc

Thank you for taking the time to answer my query - I really appreciate that.

We have two Aibo's here that the grandkids have played with as they grew up - and three battery packs. I have removed the cover off one of those - so now you have answered my query I feel confident enought to have a go at recelling the one that I have pulled the cover off. We then have two left to try to get to the U.S. if that doesn't work. But I like the idea of a new compatible cell. I have not found these on the internet so I will have another look around to see if I can find the supplier. Getting cells out of NZ for repacking is a real challenge.

I did confuse things with the readings I reported in my post. The current cell voltage measued across the outer wires (not the central one to the BMS) is 0.99volts - so the cell is flat. But the reading across the fuse was 0.91 ohms on one leg and no reading on the other (measuring from the central single leg on one sie to either of the twin legs on the other. And there is not resistance between the twin legs so it seems like one side of the thermo trigger is blown. When the replacement fuses arrive I will either attempt replacing myself or find someone to do it for me. Thanks for your advice on that.

But I am no longer confused about the TP4/PTP connections (to TP12 for protection and TP24 for reset). So apart from reconsidering whether to replace the fuse myself or not, I feel confident to attempt the recelling. Thank you for your clarification that has got me to that point. I really appreciate that - and the grandkids will hugely appreciate your help if I can sucessfully complete the recelling and breath life back into their precious friends.

I will add another post to let you know how I go with that.

Thanks. And thanks also to lvdbruno for his thread relating to desoldering etc which also helped me.

**Comment by lpollier** [comment]
I'm glad to hear you have several battery packs to work with. When I first started my re-celling tests, I only had one, and it took me multiple tries (along with a few errors and slight PCB degradation) before I really understood how everything worked. Focusing on just one pack for your first attempt, as you're planning, is definitely a good idea.

For sourcing quality cells, there are several options. For instance, Panasonic NCR18500A cells are well-known, but be cautious—there are fakes out there with less than half the advertised capacity. Another reliable provider is Vapcell; I’ve tested their INR18500 cells and measured over 2000mAh capacity. Here's one link I used, but there are other vendors on the internet:
[Aliexpress Vapcell INR18500 link](http://fr.aliexpress.com/item/1005005016158936.html?spm=a2g0o.order_list.order_list_main.10.72215e5bHHXfQ5&gatewayAdapt=glo2fra)

Now, I have already re-celled four battery packs and plan to re-cell two more using these Vapcells:
![20241116_115434](https://github.com/user-attachments/assets/5e8d9331-3a1e-4517-ad7a-9d09668c18b5)
![20241116_115529](https://github.com/user-attachments/assets/7e359698-5bd0-4213-9f1f-5088a539d8b7)

For your curiosity, I’ve been in touch with r00li regarding his project, which provides insights into equivalent battery packs. Here’s the link to his GitHub repository, where you can explore more:
[https://github.com/r00li/Aibo-ERS-xxx-battery](https://github.com/r00li/Aibo-ERS-xxx-battery)

Regarding the fuse, it seems definitely that it is half-burnt because the resistance between two of its three legs is not low (~0 to 15 ohms) but instead reads as infinite (open state). To remove it, I used a flat-tip soldering iron and carefully detached it. For soldering the new fuse, I used a fine-tip soldering iron with controlled, short-duration heat application to avoid damaging the surrounding components. This method worked well for me and might be useful for you too.

Hoping you succeed and encouraging you to take your time, exercise caution, and use the recommended tools and materials to achieve a greater success ratio. Let me know how your re-celling results turn out!

Kind,
Loïc

**Reply by GHthekiwi** [reply]
Loïc,

Great news - the battery pack that I had pulled apart to try the recelling works just fine.
It has now been through several charges and Aibo is working fine.
The granchildren are absolutely delighted to have their friend back and they have asked me to thank you very much with your help to get it going again. Without that Aibo would never have breathed life again because we simply could not afford the professional re-celling work. So from both our lovely grandkids and myself - thank you.

We have used a pair of Molicel M35A 18650 3500mAh 10A Rechargeable Batteries so the settings in the EEPROM for design and full charge capacity are AC 0D (3500). It appears that is working OK because we are getting more than 2 hours of use after a full charge. The batteries are housed in a 3D printed holder that I printed with PETG which has a softening point well above the maximum charge/discharge temperature for the batteries.

Overall - we are all very happy.

Unfortunatley the second  re celling attempt did not work out so well for some strange reason that I do not understand. I did the EEPROM resetting exactly the same as for the first one, the wiring is identical and I followed exactly the same steps. The fuse in the second pack was not blown so I did not have to replace that.

But when it was assembled, nothing was as it should have been. When I did the first one, the resistance across TP4/5 and TP24 was such that when I tested it with my multimeter the screen went blank. So I proceeded to short between TP4/5 and TP 24 and then remeasured the resistance. It was 0.
But with the second re-recelled attempt when I test the resistance between TP4/5 and TP24 my multimeter reads 1 - an open circuit with no resistance. Even more confusing, the voltage of the two Lipols measures at 1.32 volts. Each cell individually measures at 3.49 volts but when they are inserted into the box and connected to the board the combined voltage is only 1.39 volts so it seems something is terribly wrong. I assume that the BMS board is "fried"? On the first (succesful) version, the voltage of the two cells connected to the BMS after shorting TP4/5 - TP24 was 6.98 volts which is about right if they are shipped at 60% charge.

I might pull apart the third dead battery pack and build a third re-celled version unless you can shed any light on why the second attempt might be not succesful. I am just a bit nervous about destroying the last of my battery packs because perhaps I have done something wrong, and there is in fact a difference between the first and second versions. I have checked and double checked everything and I believe they are identical exceopt one works and the other doesn't.

Do you have any thoughts about what could be casuing the strange outcome on the second version?

But it is noit the end of the world - we have one working battery pack and the grandkids are deligted to reconnect with Aibo.

Thanks
Gordon

**Reply by GHthekiwi** [reply]
I did find some differences in the EEPROM settings. The was I flashed the EEPROM started with reading the EEPROM, I then adjusted the three values that need to be changed, then wrote the values tio the chip. So, everything that was written was the same as what existed on the chip except the changed values.

The comparison in the image below has circles around the differences. Could that be contributing to the strange behaviour of the BMS?
![Comparison 1st and 2nd](https://github.com/user-attachments/assets/53e59016-63a5-413e-842d-44fb1e436a30)

**Reply by GHthekiwi** [reply]
Update - I removed the batteries and cleaned all of the tracks on both sides of the board (it was pretty grubby when it came out of the battery pack and I have has problems with flux causing resistance on some of my Arduino projects). When I put the batteries back in I now get a voltage of 6.98 volts on the connected batteries. But the resistance between TP4/5 and TP24 is still 1 (infinity?) on all ranges on my multimeter (I usually use 20). Does that mean that all is now well or is it indicating that there is a problem with the board? Should I try charging it or would that not be a good move?
Cheers
Gordon

**Reply by lpollier** [reply]
Hi Gordon,

Great to hear the first recelling attempt was a success and that Aibo is back in action—happy for your grandchildren to have their friend to play with again! The details you’ve shared, especially the Molicel M35A specs and the 3D-printed PETG holder, are fantastic—thank you. I’m curious about the tools you used for reading and writing the EEPROM and if you can share the link to the 3D-printed file.

Regarding the second pack, I’d suggest focusing on resolving it before attempting a third. I can confirm the EEPROM values seem properly written, although the cycle count is showing 1 instead of 0—I assume this reflects an initial recharge attempt. Based on the bq2040 datasheet, the differing values circled in the second line likely correspond to "Manufacture date" and "Serial number" (expected to differ). The next three values ("Battery voltage offset," "Temperature offset," and "Voltage gain1") may or may not be critical. To further inspect these, I recommend the Arduino toolkit BattMon 2.0, available on that repo.

For electrical tests, check that the voltage between TP4/TP5 and TP26 (near transistor Q1) reads ~7.4V, which aligns with your two cells in series. If correct, proceed to reset the first protection level by momentarily shorting TP4/TP5 to TP24, without forgetting the usual wire connection between TP4/TP5 and TP12 to avoid frying the fuse. After this, recheck the resistance between TP4/TP5 and TP24—it should drop to a very low value, close to 0 ohms.

Let me know how it goes!

Best regards,
Loïc

**Reply by GHthekiwi** [reply]
Loïc

Thank you for taking the time to reply to me again. You are a legend - and I really appreciate the communication. I am drowning a bit and really not sure what to do next. You communication is a lifesaver thats keeping me motivated to carry on.

Anyway, first where am I at with subsequent testing. I took the board to the tech centre where I volunteer to try the measurements with a higher tech multimeter than mine.
The voltage across the two cells is about 6.98 volts and the voltage between TP4/5 and TP26 is slighly lower (about 6.7) so there is voltage at TP26.
But the resistance between TP4/5 and TP24 is still 1. I take that as an open circuit - no resistance at all. And as I understand it that is a lot different to a resistance of 0 which would be a connection with no resistance.
So, I still think there is a problem and I'm not sure what to do next !!!!

The reason that the charge counter (0000 DE) is 1 not zero is because I put 01 in there. Logic - the first charge is the new cells. It has not been in a charger yet - still in bits on my workbench. I am reluctant to put it in the charger until I understand the significance of the 1 reading on the resistance to TP24.

Do you think I should finish the assembly and try it in the charger? I am cautious because I don't understand whether that could trigger a Li Ion fire.

I'm really not sure what to test next to try and figure out why there is not close to 0 resistance on TP24. And reluctant to pull the last of my packs to bits until I have restored confidence that i have done nothing wrong.

Do you have any more ideas?

As for the rest:

I have been relying on the following links which fully disclose the resources I have been using:

1. Your github page is the one I have visited the most while I have been planning and doing the work. An excellent instructional resource.

2. The battery case is a thingiverse download -  https://www.thingiverse.com/thing:6433294 There are also excellent instructions on the page. I have not had to modify anything much. Just some simple trimming with a craft knife to ease the gap needed for the temperature strip.

3. The batteries I used can be seen at https://cell-supply.co.nz/product/molicel-m35a-18650-3500mah-10a-battery/ The alternative is Q30 cells but when I spoke to the site they recommended the Molicels. Slightly more mAh and the pairs supplied come fully balanced. And the reviews of the Molicels on their site are great.

4. The PETG filament is at https://www.3dea.co.nz/shop/product/jh-petg-clear-1-75mm/ I have a Flashforge Finder printer which does not have a heated bed but I was able to print the two parts very easily using a temperature of 230 degrees and using glue stick on the bed allowing it to dry before starting the print job.

5. The CH341A EEPROM burner is https://www.aliexpress.com/item/1005001505626412.html? Really cheap and it works fine. You need to buy the "set" to get the test clip, and the plate to connect it to the CH341A

5. If you search around the web you will learn that there is a problem with this burner - the data pins are running 5 v not 3.3 v. If you want to fix that it is really easy - best reference is https://youtu.be/HwnzzF645hA But I have not done the mod since all of the YouTube demos including burning the Aibo EEPROM have not been using a modified CH341A so I don't think the data pin issue is critical for this.

6. There are heaps of YouTube videos demonstrating how to use the CH341A. The best one is possibly https://youtu.be/Jqzp0729u_I

7. I did have many issues finding the software for the CH341A that would run under Windows 10. Most of the downloads available are infected with Trojans. But the place to go to get good installation and operation instruction is https://youtu.be/L0ChYNwunUE and that video points to the download site which is https://github.com/KrisKasprzak/CH341 KrisKasprzak has put a comprehensive install package together. If you run the setup.exe file and then the DRVSETUP64.exe file the CH341A.exe file will run seemlessly on Windows 10.

8. The battery clips that I used are at https://www.aliexpress.com/item/1005003372478123.html? I ordered the 14x12 option and they worked fine BUT I had to use a pin punch to extend the male connector a bit so it would reach the battery.

And there you have it all.
I have attached a photo of version 1 (the one that works). The cells had not been inserted when I took the photo.
If you want to try one of these but stuck for a printer I could look at printing you a couple of cases and posting them over to you.

![My version 1](https://github.com/user-attachments/assets/485544ad-ed4a-4b80-920f-1e459a5c6864)

**Reply by GHthekiwi** [reply]
PS - forgot to confirm the status of the fuse (checked again this morning).
![Fuse](https://github.com/user-attachments/assets/749c3e6d-1ecf-495c-a90e-ca1727d4ab0b)

**Reply by lpollier** [reply]
Hi Gordon,

Thank you for your kind words! And your detailed documentation, including the parts and tools you’ve used, can help others working on the battery pack recelling.

Regarding your current findings:

1. Fuse Check:
The resistance measurements being close to 0 confirm that the fuse is in good condition.

2. Battery Clips vs Spot Welding:
I noticed that you're using clips to connect the cells to the PCB. While this approach works most of the time, it can introduce potential issues. For example, shocks or vibrations might cause false contacts, which might then require triggering the first level battery protection reset again. To avoid such issues, I recommend using nickel rollers and a spot welding machine, similar to those in original battery packs. This method ensures a more secure and reliable connection.

3. Voltage Measurements:
It’s great that you’ve measured the serial cells' voltage between TP4/TP5 and TP26 of about 7V (2S voltage). To better diagnose the issue, could you also measure the voltage between TP4/TP5 and the following three test points: TP15, TP17, and TP18? I expect to see 1S voltage on TP15 and TP17 and 2S on TP18.

Best,
Loïc

**Reply by GHthekiwi** [reply]
Loïc

Thanks again for the reply. Its really great and I learn even more from you.
But - fortunately I need to worry no more and will not be needing to try these advanced measurement tests at the moment (although it will be useful in future possibly). You see, the really great news is that the second battery pack is now working OK. Last night I figured that the only thing I could do that was low risk and needed no more understanding was to re-flash the EEPROM to address the uncertainty about the criticality of differences in the EEPROM between the two packs. Fortunatley I had made a .bin copy for both cells at the start of the re-celling and again after I had flashed the EEPROM. So all I did was open the .bin file from the re-flashed version 1 which we know is working, erased the EEPROM on the second pack and wrote the EEPROM from the first working pack onto the second pack EEPROM.
Presto - everything then changed. The resistance between TP4/5 and TP24 was such that it resulted in a blank screen on my multimeter (which is what happened on the first pack) and when I shorted them the resistance reading was 0.
So, live a little, learn a little. It seems that the "Battery voltage offset," "Temperature offset," and "Voltage gain1" EEPROM settings do have a significant impact on the whole operation of the BMS. It would be good to understand why, and what all of the EEPROM settings are, but for now that is not critical because the pack is working, has fully charged OK and the Aibo runs fine off it.
So, some more knowledge about things that go wrong that anybody wanting to play with re-celling might benefit from.
I have attached a copy of the working .bin file in case somebody wants to try using it if they are similary having problems. The trick is to make sure you use the CH341A software to make a copy of the existing settings before starting any changes.
On the other suggestions you make - I am confident that the batteries will not loose contact during operation of the Aibo because the design of the printed pack leaves them very well wedged into the battery clips. Takes a lot of leverage to remove them. Besides - I do not have access to a spot welding machine.

Thanks again for all you help. Would be good to stay in touch on any other issues.
I will now start trying to figure out how I can address the DHS on the kid's second Aibo that is not working at the moment.

Gordon

[Aibo ERS210 EEPROM settings for 3500 mAh battery.zip](https://github.com/user-attachments/files/17875248/Aibo.ERS210.EEPROM.settings.for.3500.mAh.battery.zip)

**Reply by lpollier** [reply]
Hi Gordon,

Thank you for sharing the EEPROM configuration and the update on your second battery pack recelling. It’s good to hear that the pack is now working and powering your Aibo properly.

If you’re confident that the batteries are securely held by the clips, there’s no need to worry about using a spot welding machine.

That said, I’m still a bit surprised that only those EEPROM parameters (Battery voltage offset, Temperature offset, and Voltage gain1) were enough to prevent the first-level battery protection reset. My earlier analysis still suggests that this might be more related to the voltage supplying the IC1 component. When I checked the original EEPROM data I saved from my four recelled packs, I noticed slight variations in these three parameters. Specifically, I observed the following sets of values:

    (0x0B, 0x7C, 0x81)
    (0x0B, 0x7C, 0x81)
    (0x13, 0x76, 0x83)
    (0x10, 0x7B, 0x7F)

These variations suggest that the parameters can change during the pack’s operation.

Here’s part of the EEPROM memory map for the bq2040 that defines these parameters:
![image](https://github.com/user-attachments/assets/944481ab-d784-4043-8dca-9429cd5d7437)

The important thing is that your second pack is now recelled and working fine. I encourage you to perform a calibration cycle after recelling the pack: fully charge it, then fully discharge it, and finally charge it again. This can help optimize its full capacity performance.

Good luck with pinning the clutches of the head servos. For mine, I followed techniques from others in the Discord RobotCenter group, which you might find helpful. It’s a community for Aibo repairs, technical resources, and related techniques; feel free to join using this link:
https://discord.gg/xxcRs5tQ

Best regards,
Loïc
